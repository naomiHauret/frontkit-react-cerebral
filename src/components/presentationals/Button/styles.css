:root {
	--test: pink;
}

.module-style-test {
	color: var(--test);
	padding: 1.5rem;
}
