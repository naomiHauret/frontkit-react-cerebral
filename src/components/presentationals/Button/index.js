import React from "react"
import propTypes from "prop-types"
import CSSModules from "react-css-modules"
import styles from "./styles.css"

const Button = ({ label, handleClick }) => (
	<button
		data-react="rButton"
		className="bg-black sm:bg-green md:bg-yellow lg:bg-blue xl:bg-white text-red"
		styleName="module-style-test"
		onClick={handleClick}
	>
		{label}
	</button>
)

Button.propTypes = {
	label: propTypes.string,
	handleClick: propTypes.func,
}

export default CSSModules(Button, styles)
