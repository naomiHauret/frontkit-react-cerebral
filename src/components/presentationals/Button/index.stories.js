import React from "react"
import { storiesOf, addDecorator } from "@storybook/react" // eslint-disable-line no-unused-vars

// add-on
import { action } from "@storybook/addon-actions"
import { withInfo } from "@storybook/addon-info"
import { withKnobs, text } from "@storybook/addon-knobs/react"
import { withSmartKnobs } from "storybook-addon-smart-knobs"

// component
import Button from "./"

/*

// -- Won't work
// style is applied, but no knobs nor complete info

storiesOf("Button", module)
.addDecorator((story, context) => withInfo("")(story)(context))
.addDecorator(withSmartKnobs)
.addDecorator(withKnobs)
.add("With a label", () => <Button label="Hey" onClick={action("Boo")} />)

*/

// -- Work partially (knobs work, incomplete info)

storiesOf("Button", module)
	.addDecorator((story, context) => withInfo("")(story)(context))
	.addDecorator(withSmartKnobs)
	.addDecorator(withKnobs)
	.add("With a label", () => <Button label={text("label", "Hey")} onClick={action("Boo")} />)
