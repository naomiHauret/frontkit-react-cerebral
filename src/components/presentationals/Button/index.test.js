import React from "react"
import { shallow } from "enzyme"

import Button from "./"

test("render itself", () => {
	let wrapper = shallow(<Button />)
	expect(wrapper).toMatchSnapshot()
})

test("render itself with label", () => {
	let wrapper = shallow(<Button label="Hey" />)
	expect(wrapper.text()).toEqual("Hey")
})

test("click event working", () => {
	const wrapper = shallow(<Button label="Hey" handleClick={console.log("Hello world!")} />)

	expect(wrapper).toMatchSnapshot()

	wrapper.find('[data-react="rButton"]').simulate("click")

	expect(wrapper).toMatchSnapshot()
})
