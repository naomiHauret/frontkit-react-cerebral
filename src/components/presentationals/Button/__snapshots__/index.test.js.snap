// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click event working 1`] = `
<button
  className="bg-black sm:bg-green md:bg-yellow lg:bg-blue xl:bg-white text-red module-style-test"
  data-react="rButton"
>
  Hey
</button>
`;

exports[`click event working 2`] = `
<button
  className="bg-black sm:bg-green md:bg-yellow lg:bg-blue xl:bg-white text-red module-style-test"
  data-react="rButton"
>
  Hey
</button>
`;

exports[`render itself 1`] = `
<button
  className="bg-black sm:bg-green md:bg-yellow lg:bg-blue xl:bg-white text-red module-style-test"
  data-react="rButton"
/>
`;
